### PR TITLE
daemon/api_quotas.go: support updating quotas with ensure action

### PR DESCRIPTION
**Pull request by anonymouse64**
This is needed in order to support quota group updates.

### CODE REVIEW

File: daemon/api_quotas_test.go [hunk 1]
```diff
@@ -140,6 +140,46 @@ func (s *apiQuotaSuite) TestPostEnsureQuotaHappy(c *check.C) {
...
+	})
+	defer r()
+
+	updateCalled := 0
+	r = daemon.MockServicestateUpdateQuota(func(st *state.State, name string, opts servicestate.QuotaGroupUpdate) error {
+		updateCalled++
+		c.Assert(name, check.Equals, "ginger-ale")
+		c.Assert(opts, check.DeepEquals, servicestate.QuotaGroupUpdate{
+			AddSnaps:       []string{"some-snap"},
+			NewMemoryLimit: 1000,
```
Comment (pedronis) [inline daemon/api_quotas_test.go:162]: should we make this larger for realism given that atm we don't support reducing memory?

**Review by pedronis — APPROVED** [review]
thanks, one detail about the test

File: daemon/api_quotas_test.go (hunk 1 above)
Comment (anonymouse64) [reply daemon/api_quotas_test.go:162]: ah good point yes that is a good point

**Review by bboozzoo — APPROVED** [review]
LGTM, thank you!

**Pull request merged**